alfasim_sdk/context.py

```diff
@@ -207,87 +207,232 @@ class Context:
     and the models implemented by the user.
+
+    The following methods provide an instance of :func:`~alfasim_sdk.context.Context` to inform the current state
+    of the application:
+
+    - :ref:`visible-expression-section` parameter from all fields
+    - :ref:`enable-expression-section` parameter from all fields
+    - :func:`~alfasim_sdk.hook_specs_gui.alfasim_get_status` hook
     """
 
     def GetModel(self, model_name: str) -> Optional[type]:
         """
-        Returns a model defined from a plugin.
-        The parameter ``model_name`` must be the name of models defined within the plugin.
+        Returns an instance of the given ``model_name``.
+
+        The parameter ``model_name`` must be the name of a model defined within the plugin.
+
+        In the example below, the Context is used to access a property from the model ``MyModel``
+
+        ``ctx.GetModel("Acme")`` as exemplified in the code below.
+
+        .. rubric:: Setting up the model
+
+        .. code-block:: python
+
+            @data_model(caption="MyPlugin", icon="")
+            class MyModel:
+                name = String(value="ALFAsim", caption="Field")
+                scalar = Quantity(value=1, unit="degC", caption="Field")
 
-        Ex.:
-        You can access a class "Acme" defined from ``alfasim_get_data_model_type()`` by typing
-        ``ctx.GetModel("Acme")`` as exemplified in the code bellow.
+            @alfasim_sdk.hookimpl
+            def alfasim_get_data_model_type():
+                return [MyModel]
 
+        .. rubric:: Accessing the context
 
-        @hookimpl
-        def alfasim_get_data_model_type():
-            @model_class(caption="Acme")
-            class Acme:
-                name = String(value="acme", caption="Acme")
+        .. code-block:: console
 
-            return [Acme]
 
-        @hookimpl
-        def alfasim_get_status(ctx):
-            model = ctx.GetModel('Acme')
-            assert model.name == 'acme'
+            >>> ctx.GetModel('MyModel')
+            MyModel(name='ALFAsim', scalar=Scalar(1.0, 'degC', 'temperature'))
 
-        It also possible to use ``alfasim_sdk.context.PluginInfo.models`` to check models available for a given Plugin
-        For more information check ``alfasim_sdk.context.Context.GetPluginInfoById``
+            >>> ctx.GetModel('MyModel').name
+            'ALFAsim'
+
+        At runtime, you can also verify the names of the models defined by a given plugin. For this, you need to
+        For more information check :func:`~alfasim_sdk.context.Context.GetPluginInfoById`
+
+        :raises TypeError: When the given ``model_name`` does not exist.
+        :raises FrozenInstanceError: When trying to modify a value
 
-        The values from the returned Model are read-only, they cannot be modified.
-        If the model informed cannot be found, a  :class:`TypeError` exception is raised.
         """
 
     def GetPipelines(self) -> Optional[List[PipelineInfo]]:
```

Comment:
Just noticed... how are **wells** being handled? They are special nodes with edge information, not sure how the plugins are seeing that.

Reply:
All the context infrastructure and code were built before the functionality of the well. So the plugins don't have any information about the wells